akka/src/main/scala/auth/handlers/AuthActionsHandler.scala

```diff
@@ -56,94 +62,62 @@ class AuthActionsHandler(service: AuthService, emailPasswordServices: EmailPassw
-    implicit request ⇒
-      (for {
-        pid ← passwordRecoveryService.finishRecovery(request.body.token.email, request.body.json.newPass)
-        aStatus ← authStatus(pid, false)
-      } yield aStatus.ok) recoverWith {
-        case e: JsonApiFailure ⇒ e.copy(status = 401).resultF
-      }
-  }
-           */
-          complete("???")
-        } ~ path ("verifyEmail") {
-          /*
-          Action.async(authToken[EmailVerifyToken, EmailVerifyCommand]) {
-    implicit request ⇒
-      (for {
-        result ← emailVerifierService.verify(request.body.email)
-      } yield NoContent) recover {
-        case e: JsonApiFailure ⇒ e.result
-        case u: Exception      ⇒ JsonApiFailure(400, "cannot_verify_email", u.getMessage, "auth").result
-      }
-  }
-           */
-          complete("???")
-        } ~ (path ("requestEmailVerify") & userRequired) { s ⇒
+        } ~ (path("startPasswordRecovery") & entity(as[StartPasswordRecover])) { spr =>
+          onSuccess(passwordRecoveryService.startRecovery(spr.email)) {
+            complete(emptyResponse)
+          }
+        } ~ (path("checkPasswordRecovery") & authTokenCommand[PasswordRecoverCommand, CheckPasswordRecoverToken]) { _ =>
+          complete(emptyResponse)
+        } ~ (path("recoverPassword") & authTokenExpirableCommand[PasswordRecoverCommand, FinishPasswordRecover]()) { cmd =>
+          onSuccess(passwordRecoveryService.finishRecovery(cmd._1.email, cmd._2.newPass).flatMap(authService.getStatus)) { authStatus =>
+            respondWithAuth(authStatus) {
+              complete(authStatus)
+            }
+          }
+        } ~ (path("verifyEmail") & authTokenCommand[EmailVerifyCommand, EmailVerifyToken]) { cmd =>
+          onSuccess(emailVerifierService.verify(cmd._1.email)) {
+            complete(emptyResponse)
+          }
+        } ~ (path("requestEmailVerify") & userRequired) { s ⇒
           complete(
             emailVerifierService.startVerify(s.userId)
-              .map(_ ⇒ StatusCodes.NoContent → HttpEntity.empty(ContentTypes.NoContentType))
+              .map(_ ⇒ emptyResponse)
           )
-        } ~ path ("checkEmailAvailability") {
-          /*
-          Action.async(authBodyJson[EmailCheckAvailability]) {
-    implicit request ⇒
-      profileRegistrar
-        .isEmailRegistered(request.body.email).map {
-          case true ⇒
-            JsonApiFailure(400, "email_already_registered", "Email already registered", "auth").result
-          case false ⇒
-            NoContent
+        } ~ (path("checkEmailAvailability") & entity(as[EmailCheckAvailability])) { eca =>
+          onSuccess(service.isEmailRegistered(eca.email)) {
+            case true => complete(emptyResponse)
+            case false => failWith(AuthError.UserAlreadyRegistered)
+          }
+        } ~ (path("startEmailChange") & userRequired & entity(as[StartEmailChange])) { (status, cmd) =>
+          onSuccess(Future(emailChangeService.start(status.userId, cmd.email))) {
```

Comment:
probably some change in `emailChangeService` is needed? or is it completely useless?